client/src/components/Questions/QuestionsList/Answer/Answer.jsx

```diff
@@ -3,26 +3,61 @@ import PropTypes from 'prop-types';
 import Button from './Button/Button';
 import style from './Answer.css';
 
-function Answer({ answer }) {
-  return (
-    <div className={style.answer}>
+// make yes/no buttons increment count
+//   store counts in local state
+//   implement click handler that updates local state
+//
+//   stretch: update database
 
-      <div className={style.user}>
-        <span>{`${answer.user.nickname} · x days ago`}</span>
-      </div>
+class Answer extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      yes: 0,
+      no: 0,
+    };
+    this.handleClick = this.handleClick.bind(this);
+  }
 
-      <div className={style.text}>
-        <span>{answer.text}</span>
-      </div>
+  componentDidMount() {
+    const { answer } = this.props;
+    this.setState({
+      yes: answer.useful.yes,
+      no: answer.useful.no,
+    });
+  }
+
+  handleClick(e, id) {
+    // this.setState((prevState) => ({ yes: prevState.yes + 1 }));
+    // this.setState((prevState) => ({ no: prevState.no + 1 }));
+    e.preventDefault();
+    this.setState((prevState) => ({ [id]: prevState[id] + 1 }));
```

Comment:
I don’t see id in the state. I would love to know how this setState works :) teach me